webpack/webpack-embedded-dependencies-plugin/src/index.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import path from 'path';
+import type { IOptions } from 'glob';
+
+import { FileSystem, Import, LegacyAdapters } from '@rushstack/node-core-library';
+import { Compilation, Compiler, WebpackPluginInstance, sources } from 'webpack';
+
+const glob: typeof import('glob') = Import.lazy('glob', require);
+const globAsync = (pattern: string, options: IOptions = {}): Promise<string[]> => {
+  return LegacyAdapters.convertCallbackToPromise(glob, pattern, options);
+};
+
+const COPYRIGHT_REGEX: RegExp = /^Copyright .*$/m;
```

Comment:
Put a comment with an example of what this matches?

Reply:
After running it through these tests, it appears that this RegExp is only successfully capturing matches inside of MIT-like licenses. I will need to expand it to support the other licenses I've provided in fixtures.

Reply:
Added comments for these regular expressions. They have also been isolated to a separate file and exported so that they are not exposed from the plugin export but still testable.